### DISCUSSION TITLE
Remote Server

### DISCUSSION THREAD

**Discussion by trymeouteh** [podman-desktop/podman-desktop#5526] [Ideas]
Be able to use podman desktop to connect to a remote machine and control the podman cli on that machine and manage it in podman desktop remotely.

For exmaple, you have two machines. One if your laptop, and the other being a server. You install podman desktop on the laptop and install podman cli on the server. You can use podman desktop to connect to the server and control podman on that server in podman desktop!

And be able to manage more than one server. Have a panel on the side which will list your local device and any remote servers you have saved.

- Local Device
- Server 1
- Server 2

https://github.com/containers/podman-desktop/issues/279#issuecomment-1817763216

**Comment by afbjorklund** [comment]
Since Podman Desktop doesn't support either `tcp:` or `ssh:`, you will have to do the tunneling yourself and use `unix:`.

Of course it doesn't support setting the socket path either, so you would have to "abuse" one of the existing providers.

**Reply by afbjorklund** [reply]
Here is the feature for Lima, to be able to add any existing server as a linux machine (lima) instead of starting a VM:

* https://github.com/lima-vm/lima/issues/2109

It handles the ssh connection ,and the port forwarding (including the unix socket) and filesystem mounting for you.

And then you can just use Podman Desktop with the Lima provider, to talk to the `docker.sock` or `podman.sock`...

**Comment by afbjorklund** [comment]
It is possible to use more than one Lima, but currently the Podman Desktop only has single fields for "the" instance.

![podman-desktop-lima-prefs](https://github.com/containers/podman-desktop/assets/10364051/1c3d2a7e-b4c9-4a97-a524-25b70e617436)

For it to work with more than one (at the same time), it would have to be converted to a _list_ of connections first.
It is the same situation as when you want to run **both** Podman and Kubernetes, but there is another hack for that.
(currently it is a drop-down, so you can't enable both at the same time. and not have different settings for each either)

* https://github.com/containers/podman-desktop/issues/5206

The server doesn't need the podman CLI (unless you prefer to call that remotely), it is enough with the "daemon" API.
The "podman system connection" also allows you to configure more than one. Podman Desktop could use that?
(for some reason, the list doesn't include the local daemon unlike the "docker context" which shows both of them)

`podman --remote system connection ls`

* https://podman-desktop.io/docs/podman/setting-podman-machine-default-connection

**Comment by trymeouteh** [comment]
Adding a SSH client inside of podman desktop would be a good solution. SSH allows for an encrypted channel. To connect to a remote device in podman desktop, the user can connect to the remote device using SSH within podman desktop.

**Reply by afbjorklund** [reply]
Podman Desktop does **not** need a SSH client, it runs `podman machine ssh` or `lima`, and then those are talking to the remote machine. If they need an external `ssh` client, or if they rely on the embedded Go connection, that is up to those installers...

But _normally_ you would only need `run` and `exec`, to talk to containers.

Podman Desktop instead supplies a web terminal, so that you can interact with those from the web application that it is: (now that PD is doing more and more "package management", it would be _possible_ to install OpenSSH with it - but that's secondary)

* #4493

**Comment by ghost** [comment]
using powershell's ability to handle the wsl mount I manage my podman-machine from anywhere in my house:

```
cd \\wsl.localhost\podman-machine-default\mnt\wsl\podman-sockets\podman-machine-default ;
New-Item -ItemType SymbolicLink -Path "C:\Users\ME\podman-root.sock" -Target ".\podman-root.sock"
```
-manually add id_ed25519.pub to podman-machine-default/root/.ssh/authorized_keys

```
ssh -L 44444:127.0.0.1:65157 ME@windowsmachinerunningPodmanDesktop -N > /dev/null 2>&1 &
```
check with ssh localhost -p 44444 -l root

```
podman system connection add --identity ~/.ssh/id_ed25519 -d windowsmachinerunningPodmanDesktop ssh://root@localhost:44444
```

**Comment by RushingAlien** [comment]
> Since Podman Desktop doesn't support either tcp: or ssh:

This is no longer true,
```
podman system connection add connection-name ssh://address:port/path/to/socket
```
then doing
```
podman -c connection-name options-here
```
works

**Reply by afbjorklund** [reply]
Those look like CLI commands, rather than the GUI configuration discussed here

**Reply by afbjorklund** [reply]
Now Podman Desktop (GUI) handles the ssh connection as well, since PR #8265
